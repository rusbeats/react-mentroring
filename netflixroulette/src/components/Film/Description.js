import React from 'react';
// Hardcoding the value only for homework #3
const filmDescription = 'The Matrix is a 1999 science fiction action'
                + ' film written and directed by The Wachowskis and'
                 + ' starring Keanu Reeves, Laurence Fishburne, Carrie-Anne Moss,'
                  + ' Hugo Weaving, and Joe Pantoliano. It depicts a dystopian future'
                  + ' in which humanity is unknowingly trapped inside a simulated reality'
                   + ' called the Matrix, created by thought-capable machines (artificial beings)'
                    + ' to control humans while using their bodies as an energy source.';

const Description = () => (
  <div className="film-desc">
    {filmDescription}
  </div>
);
export default Description;
