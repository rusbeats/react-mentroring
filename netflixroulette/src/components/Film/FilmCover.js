/* eslint-disable jsx-a11y/alt-text */
/* eslint-disable react/destructuring-assignment */
import React from 'react';

const FilmCover = props => (
  <div className="film-cover">
    <img className="film-image" src={props.filmImage} />
  </div>
);
export default FilmCover;
