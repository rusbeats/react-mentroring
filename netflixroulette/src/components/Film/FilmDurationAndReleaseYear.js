import React from 'react';
// Hardcoding the value only for homework #3
const duration = '152 min';
const releaseYear = '2002';

const FilmDurationAndReleaseYear = () =>(
            <div className="film-duration-and-release-year">
                <span>{duration} &nbsp; {releaseYear}</span>
            </div>
    );
export default FilmDurationAndReleaseYear;

