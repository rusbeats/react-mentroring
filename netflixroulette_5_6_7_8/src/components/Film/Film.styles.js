import constants from '../../utils/constants';
import styled from 'styled-components';

const { colors } = constants;

export const FilmTitleAndDateWrapper = styled.div`
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 10px 0;
`;

export const FilmTitle = styled.div`
    text-transform: uppercase;
    font-size: 20px;
    font-weight: bold;
    max-width: 250px;
`;

export const FilmDate = styled.div`
    border-style: solid;
    border-width: 1.5px;
    border-color: ${colors.grey};
    border-radius: 3px;
    padding: 3px 10px;
`;

export const FilmGenres = styled.div`
    color: ${colors.grey};
    max-width: 350px;
`;
