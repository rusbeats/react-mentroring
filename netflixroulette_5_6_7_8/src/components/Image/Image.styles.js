import styled from 'styled-components';

export const Image = styled.img.attrs(props => ({ alt: '', src: props.poster_path }))`
    height: 500px;
    width: 350px;
`;
