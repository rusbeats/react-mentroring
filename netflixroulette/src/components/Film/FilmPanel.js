import React from 'react';
import FilmTitle from './FilmTitle';
import FilmCover from './FilmCover';
import FilmDurationAndReleaseYear from './FilmDurationAndReleaseYear';
import FilmRating from './FilmRating';
import Description from './Description';
import SearchButton from './SearchButton';

const FilmPanel = () => (
  <div className="film-panel">
    <FilmCover filmImage="../../src/img/matrix.jpg" />
    <div className="film-info">
      <div className="film-title-and-rating">
        <FilmTitle />
        {' '}
        <FilmRating />
      </div>
      <FilmDurationAndReleaseYear filmDuration="152 min" filmReleaseYear="2002" />
      <Description />
    </div>
    <SearchButton />
  </div>
);
export default FilmPanel;
