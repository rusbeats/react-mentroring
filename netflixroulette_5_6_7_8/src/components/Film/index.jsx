import React from 'react';
import PropTypes from 'prop-types';
import { getYear } from '../../utils/helpers';
import { Image } from '../Image/Image.styles';
import { FilmTitleAndDateWrapper, FilmTitle, FilmDate, FilmGenres } from './Film.styles';

const Film = ({ film, onClick }) => {
    const releaseYear = getYear(film.release_date);
    const genresLength = film.genres.length;

    return (
        <div onClick={onClick}>
            <Image src={film.poster_path} />
            <FilmTitleAndDateWrapper>
                <FilmTitle>{film.title}</FilmTitle>
                <FilmDate>{releaseYear}</FilmDate>
            </FilmTitleAndDateWrapper>
            <FilmGenres>
                {film.genres.map((genre, index) => (index === genresLength - 1 ? genre : genre + ' & '))}
            </FilmGenres>
        </div>
    );
};

Film.propTypes = {
    film: PropTypes.object,
    onClick: PropTypes.func
};

export default Film;
