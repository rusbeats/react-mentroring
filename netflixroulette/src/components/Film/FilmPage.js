import React from 'react';
import FilmPanel from './FilmPanel';
import FilmByTheSameGenre from '../Item/FilmByTheSameGenre';

const FilmPage = () => (
  <div className="film-page">
    <FilmPanel />
    <FilmByTheSameGenre />
  </div>
);
export default FilmPage;
