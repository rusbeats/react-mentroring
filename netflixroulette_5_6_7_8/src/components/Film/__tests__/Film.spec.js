import React from 'react';
import { shallow } from 'enzyme';
import Film from '../index';
import { Image } from '../../Image/Image.styles';

describe('Film', () => {
    it('should render correctly (snapshot)', () => {
        const film = {
            poster_path: 'http://some-url.com',
            title: 'Test title',
            release_date: '2000-12-31',
            genres: ['one', 'two', 'three']
        };

        const component = shallow(<Film film={film} />);
        expect(component).toMatchSnapshot();
    });

    it('should have correct src prop', () => {
        const src = 'https://some-url.com';
        const component = shallow(<Image alt="" src={src} />);
        expect(component.props().src).toEqual(src);
    });
});
