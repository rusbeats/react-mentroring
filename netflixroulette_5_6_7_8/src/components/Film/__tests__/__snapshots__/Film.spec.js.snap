// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film should render correctly (snapshot) 1`] = `
<div>
  <Imagestyles__Image
    src="http://some-url.com"
  />
  <Filmstyles__FilmTitleAndDateWrapper>
    <Filmstyles__FilmTitle>
      Test title
    </Filmstyles__FilmTitle>
    <Filmstyles__FilmDate>
      2000
    </Filmstyles__FilmDate>
  </Filmstyles__FilmTitleAndDateWrapper>
  <Filmstyles__FilmGenres>
    one & 
    two & 
    three
  </Filmstyles__FilmGenres>
</div>
`;
